feat(network): Add spike_train_drive for explicit inter-network inputs

Myrausman opened the pull request:
This PR introduces a new method `Network.add_spike_train_drive()` to enable driving a network with explicitly defined spike trains. This feature is primarily aimed at facilitating one-way communication between two simulated networks (e.g., using the spike output of a source network to drive a target network) or for incorporating externally generated/recorded spike data as inputs.

How it Works (Conceptual):

- A user simulates a source network (Network A) and extracts its spike output.
- This output is formatted into [spike_data](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) and [target_config](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).
- A target network (Network B) is initialized.
- [net_B.add_spike_train_drive()](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) is called with the formatted data to configure Network B to receive these spikes as inputs.

asoplata reviewed (commented):
I know I said this in the meeting but just to make mention again, in general, we want to be instantiating our network using `jones_2009_model()` similar to that in the following example https://jonescompneurolab.github.io/textbook/content/08_using_hnn_api/plot_firing_pattern.html . This important function does make some small changes to the network that are not present purely in the `default.json` file, and that's the version of the model we want to make sure always works in every situation.

asoplata commented on hnn_connect_networks.py:
```diff
@@ -0,0 +1,100 @@
+import csv
+
+# Network A
+params_A = read_params(op.join(hnn_core.__path__[0], 'param', 'default.json'))
```
whoops accidentally pasted two versions of this same comment

Myrausman commented:
Hi @asoplata ,

I've implemented add_spike_train_drive for network communication, using jones_2009_model(). It handles spike data transmission. Key changes:

- Added add_spike_train_drive to Network class
- Supported spike_train in _drive_cell_event_times
- Created example in hnn_connect_networks.py

Could you please review the output as well:
![image](https://github.com/user-attachments/assets/a01fc396-ca97-4715-8b78-d47bb2866903)

Thank you!

Best,
Maira

asoplata commented:
Comments and code review discussion can be found on this week's Notion page here: https://www.notion.so/jonescompneurolab/Maira-Weekly-Meeting-2025-06-18-21644dbbcce6808b824ae34346025a05
